RUFAS/output_manager.py

```diff
@@ -2200,3 +2201,550 @@ def run_startup_sequence(
+            "filter_by_exclusion": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "customization_details": self.validate_customization_details,
+            "legend": self.validate_list_of_strings,
+            "display_units": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "omit_legend_prefix": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "omit_legend_suffix": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "expand_data": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "use_fill_value_in_gaps": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "use_fill_value_at_end": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "mask_values": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "use_calendar_dates": partial(self.validate_type, expected=bool, type_label="a boolean"),
+            "data_significant_digits": partial(self.validate_type, expected=int, type_label="an integer"),
+            "title": partial(self.validate_type, expected=str, type_label="a string"),
+            "slice_start": partial(self.validate_type, expected=int, type_label="an integer"),
+            "slice_end": partial(self.validate_type, expected=int, type_label="an integer"),
+        }
+
+        for key, value in value.items():
+            if key == "date_format":
+                Utility.validate_date_format(value)
+                continue
+            if key == "fill_value":
+                continue
+            if key not in key_validators:
+                self.add_error(
+                    "Unknown key in graph details.",
+                    f"Key: {key}, is not a supported option in {filter_name}.",
+                    info_map,
+                )
+                continue
+            key_validators[key](value, key, filter_name)
+
+    def validate_type(self, value: Any, content_name: str, filter_name: str, expected: type, type_label: str) -> None:
+        """
+        Generic type checker.
+
+        Parameters
+        ----------
+        value : Any
+            The value to check.
+        content_name : str
+            Name of the field, for error messages.
+        filter_name: str
+            Name of the filter validated.
+        expected : type
+            A type or tuple of types that value must be an instance of.
+        type_label : str
+            A human-readable description of the type (used in the error message).
+        """
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self.validate_type.__name__,
+        }
+        if not isinstance(value, expected):
+            self.add_error(
+                "Invalid report filter data type.",
+                f"[ERROR] '{content_name}' in {filter_name} " f"must be {type_label}.",
```

Comment:
Would it be possible to add the actual received type?
```suggestion
                f"[ERROR] '{content_name}' in {filter_name} " f"must be {type_label} but received {content_type}.",
```